src/protocol/binary_packet.rs

```diff
@@ -121,6 +121,8 @@ pub fn parse_cas_response<R: io::Read>(stream: &mut R) -> Result<bool, MemcacheE
 pub fn parse_header_only_response<R: io::Read>(reader: &mut R) -> Result<(), MemcacheError> {
     let header = PacketHeader::read(reader)?;
     if header.vbucket_id_or_status != ResponseStatus::NoError as u16 {
+        let mut buffer = vec![0; header.total_body_length as usize];
```

Comment:
Let's move this outside of the `if`? If  `extras_length != 0` and `ResponseStatus == NoError`, we might still leave out bytes on the socket.